File: Tests/FakeServices/FakeCallService.m
//
//  FakeCallService.m
//  QBRTCDemo
//
//  Created by Anton Sokolchenko on 1/26/16.
//  Copyright © 2016 anton. All rights reserved.
//

#import "FakeCallService.h"
#import "FakeSignalingChannel.h"


#import "CallServiceProtocol.h"
#import "CallServiceDataChannelAdditionsDelegate.h"
#import "CallServiceDelegate.h"

#import <RTCPeerConnectionFactory.h>
#import <RTCPeerConnectionInterface.h>
#import <RTCMediaConstraints.h>
#import <RTCVideoSource.h>
#import "SVSignalingChannelState.h"

@interface FakeCallService()
@property (nonatomic, strong) id<SVSignalingChannelProtocol> signalingChannel;
@end

@implementation FakeCallService

- (void)startCallWithOpponent:(SVUser *)opponent {

	RTCPeerConnectionFactory *factory = [[RTCPeerConnectionFactory alloc] init];
	
	RTCVideoTrack *emptyVideoTrack = [[RTCVideoTrack alloc] initWithFactory:factory source:[RTCVideoSource new] trackId:@"trackID"];
	
	if ([self.multicastDelegate respondsToSelector:@selector(callService:didReceiveLocalVideoTrack:)]) {
		[self.multicastDelegate callService:self didReceiveLocalVideoTrack:emptyVideoTrack];
	}
	
	if ([self.multicastDelegate respondsToSelector:@selector(callService:didReceiveRemoteVideoTrack:)]) {
		[self.multicastDelegate callService:self didReceiveRemoteVideoTrack:emptyVideoTrack];
	}
	
	if ([self.multicastDataChannelDelegate respondsToSelector:@selector(callService:didOpenDataChannel:)]) {
		[self.multicastDataChannelDelegate callService:self didOpenDataChannel:nil];
	}
}

- (BOOL)isInitiator {
	return YES;
}

- (BOOL)isDataChannelReady {
	return YES;
}

- (BOOL)sendText:(NSString *)text {
	return YES;
}

- (BOOL)sendData:(NSData *)data {
	return YES;
}

@end

File: Tests/FakeServices/FakeCallService.swift
//
//  FakeCallService.swift
//  QBRTCDemo
//
//  Created by Anton Sokolchenko on 09.12.16.
//  Copyright © 2016 anton. All rights reserved.
//

import Foundation


#if QBRTCDemo_s
	@testable
	import QBRTCDemo_s
#elseif QBRTCDemo
	@testable
	import QBRTCDemo
#endif


class FakeCallService: CallService {
	
	var shouldBeConnected = false
	var shouldBeConnecting = false

	override var isConnected: Bool {
		return shouldBeConnected
	}
	
	override var isConnecting: Bool {
		return shouldBeConnecting
	}
	
	override func startCallWithOpponent(_ user: SVUser) throws {
		let factory = RTCPeerConnectionFactory()

		let videoSource = factory.videoSource()
		let emptyVideoTrack = factory.videoTrack(with: videoSource, trackId: "trackID")
		
		observers |> { $0.callService(self, didReceiveLocalVideoTrack: emptyVideoTrack) }
		
		observers |> { $0.callService(self, didReceiveRemoteVideoTrack: emptyVideoTrack) }
	}
	
}
